### ISSUE TITLE
First batch of PRs

### ISSUE THREAD

**Issue by bstaletic** [YannickJadoul/pybind11-cleanup#1]
I don't feel comfortable pushing to master, so I'll post a list of what I've gone through so far.

- pybind/pybind11#781 - Far from finished. Seems very useful. I tried finishing it, but got stuck. Having a "lock into `py::class_` after `.def`" makes it hard to get done.
  - Proposed category: Major
- pybind/pybind11#1043 - Something about propagation of return policies. Removes 4 lines, should be safe, but there's a possibility that tests just don't cover what those 4 lines are doing.
  - Proposed category: Fixes
- pybind/pybind11#1084 - Bug fix. Reviewed - looks fine to merge.
  - Proposed category: Fixes
- pybind/pybind11#1089 - Cmake C++ standard version settings. Turned into bikeshedding about cmake versions and whether cmake 3.3 is too new (in 2017). Good candidate for merge.
  - Proposed category: Minor
- pybind/pybind11#1098 - Cmake compiler features, by @henryiii. I haven't given it a proper review.
  - Proposed category: To review
- pybind/pybind11#1101 - Slicing with None. Simple enough, but blocked on repetition of code because `cast()` is not available and @henryiii claims forward declaration to be tricky.
  - Proposed category: Minor
- pybind/pybind11#1122 - Construct `py::enum_` from `std::string`. A simple feature turned into a discussion about case sensitivity. I'm questioning the usefulness of the PR.
  - Proposed category: To review (bikeshed)
- pybind/pybind11#1131 - Override overload order with `py::prepend()`. Jakob raised a concern about performance, but without backing it with numbers. Otherwise, no complaints and ready to be merged.
  - Proposed category: Major
- pybind/pybind11#1138 - Inconsistent use of holder classes mess. Seems superseded by #1161. Either this or #1161 should get a more detailed review, before categorizing.
  - Proposed category: To close
- pybind/pybind11#1161 - Look at #1138
  - Proposed category: To review

**Comment by henryiii** [comment]
Can't you make this at least a PR? There's nothing in master. And having to copy/paste your comments in kind of defeats the purpose.

A few notes:

* For 1043: tests were added, this fixes a bug that previously was not part of the tests.
* For 1101, I think with some time and maybe some help I could get the declaration right. I've marked it WIP.
* For 1122: I think it needs to be based on whatever we do with enums. Having a string constructor for enums enables a user to turn on implicit conversion from strings, which would be very useful.
* For 1131: Jakob didn't understand the PR in the first comment, once he did (in the second comment) he didn't have any performance concerns. It was approved but never merged.
